From f8084a4e6e879d89d3aa69d197293e4a85429dfb Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 20 Dec 2023 18:32:03 +0100
Subject: GCC: do not pass Clang specific MTE flags

-Xclang parameters break GCC build as they are not supported in GCC.

Bug: 819294
Change-Id: I421bd6bdbf0da57ed26ada6caa7b26831a49fc01
---
 .../partition_allocator/src/partition_alloc/BUILD.gn          | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/allocator/partition_allocator/src/partition_alloc/BUILD.gn b/base/allocator/partition_allocator/src/partition_alloc/BUILD.gn
index f3e9c4c00fb72..1516605a10caf 100644
--- a/base/allocator/partition_allocator/src/partition_alloc/BUILD.gn
+++ b/base/allocator/partition_allocator/src/partition_alloc/BUILD.gn
@@ -256,7 +256,7 @@ if (is_clang_or_gcc) {
   }
 
   config("memory_tagging") {
-    if (current_cpu == "arm64" &&
+    if (current_cpu == "arm64" && is_clang &&
         (is_linux || is_chromeos || is_android || is_fuchsia)) {
       # base/ has access to the MTE intrinsics because it needs to use them,
       # but they're not backwards compatible. Use base::CPU::has_mte()
@@ -938,7 +938,7 @@ if (build_with_chromium) {
       }
     }
 
-    if ((is_android || is_linux) && target_cpu == "arm64") {
+    if ((is_android || is_linux) && is_clang && target_cpu == "arm64") {
       cflags = [
         "-Xclang",
         "-target-feature",
-- 
2.43.0

